search/search-client-query-transformer/src/main/java/io/camunda/search/clients/transformers/query/SearchQueryResultTransformer.java

```diff
@@ -16,30 +16,31 @@
 import java.util.List;
 import java.util.stream.Collectors;
 
-public final class SearchQueryResultTransformer<T, R>
-    implements ServiceTransformer<SearchQueryResponse<T>, SearchQueryResult<R>> {
+public final class SearchQueryResultTransformer<T, R> {
   final ServiceTransformer<T, R> documentToEntityMapper;
 
   public SearchQueryResultTransformer(final ServiceTransformer<T, R> documentToEntityMapper) {
     this.documentToEntityMapper = documentToEntityMapper;
   }
 
-  @Override
-  public SearchQueryResult<R> apply(final SearchQueryResponse<T> value) {
-    final var hits = value.hits();
+  public SearchQueryResult<R> apply(final SearchQueryResponse<T> value, final boolean reverse) {
+    final var hits = reverse ? value.hits().reversed() : value.hits();
```

Comment:
❓ Maybe I overlooked it, but can you explain why this reversing is needed? There is no explanation in the commit, code, PR, or issue description. It is not an obvious change and should be documented if possible.

Reply:
yes sure, it is related to https://camunda.slack.com/archives/C06UKS51QV9/p1734115692563139 . I will add a comment explaining it

Reply:
Thanks! That should be helpful for anyone coming across this in the future.

Reply:
done 👍

Reply:
Really good explanation, this is DEFINITELY going to help in the future! 🚀